crates/ruma-macros/src/events/event_content.rs

```diff
@@ -797,9 +990,22 @@ fn generate_static_event_content_impl(
     }
 }
 
-fn needs_redacted(is_custom: bool, event_kind: Option<EventKind>) -> bool {
+fn needs_redacted(is_custom_redacted: bool, event_kind: Option<EventKind>) -> bool {
     // `is_custom` means that the content struct does not need a generated
     // redacted struct also. If no `custom_redacted` attrs are found the content
     // needs a redacted struct generated.
-    !is_custom && matches!(event_kind, Some(EventKind::MessageLike) | Some(EventKind::State))
+    !is_custom_redacted
+        && matches!(event_kind, Some(EventKind::MessageLike) | Some(EventKind::State))
+}
+
+fn needs_possibly_redacted(
+    is_custom_redacted: bool,
+    is_custom_possibly_redacted: bool,
+    event_kind: Option<EventKind>,
+) -> bool {
+    // `is_custom_possibly_redacted` means that the content struct does not need
+    // a generated possibly redacted struct also. If no `custom_redacted` or
+    // `custom_possibly_redacted` attrs are found the content needs a redacted
+    // struct generated.
+    !is_custom_redacted && !is_custom_possibly_redacted && event_kind == Some(EventKind::State)
```

Comment:
```suggestion
    !is_custom_possibly_redacted && event_kind == Some(EventKind::State)
```

based on what I've seen above, everywhere we use `custom_redacted`, we could still actually use the generated `PossiblyRedacted` type, right?

Reply:
No, for `RoomMemberEventContent` we'd need a custom `PossiblyRedacted`. So in practice there's one case where it would work and one case where it wouldn't work.

Reply:
Okay, let's keep the current implementation then I guess.

Reply:
You know what, I'm actually gonna change this. It's less annoying right now because I won't need to implement a custom struct and its traits. :stuck_out_tongue: